core/domain/draft_upgrade_services.py

```diff
@@ -164,6 +164,31 @@ def extract_html_from_draft_change_list(draft_change_list):
 class DraftUpgradeUtil(python_utils.OBJECT):
     """Wrapper class that contains util functions to upgrade drafts."""
 
+    @classmethod
+    def _convert_states_v34_dict_to_v35_dict(cls, draft_change_list):
+        """Converts draft change list from state version 34 to 35. State
+        version 35 upgrades all explorations that use the MathExpressionInput
+        interaction to use one of AlgebraicExpressionInput,
+        NumericExpressionInput, or MathEquationInput interactions. There should
+        be no changes to the draft for this migration.
+
+        Args:
+            draft_change_list: list(ExplorationChange). The list of
+                ExplorationChange domain objects to upgrade.
+
+        Returns:
+            list(ExplorationChange). The converted draft_change_list.
+        """
+        for change in draft_change_list:
+            # We don't want to migrate any changes that involve the
+                # MathExpressionInput interaction.
+            if (change.property_name ==
+                    exp_domain.STATE_PROPERTY_INTERACTION_ID and (
+                        change.new_value == 'MathExpressionInput')):
+                raise Exception('Conversion cannot be completed.')
```

Comment:
Where is the try catch where  the conversion function is called that catches this error and returns None?

Reply:
Added.